plugins/kafka-info/src/components/KafkaInfoComponent/KafkaInfoComponent.tsx

```diff
@@ -1,118 +1,198 @@
+      <InfoCard title="Kafka Info">
+        <Typography variant="body1">{"Error fetching topic data: ${message}"}</Typography>
+      </InfoCard>
+    )
+  }
+
+  if (!topics || !topics.items || topics.items.length ===0) {
+    return (
+      <InfoCard title="Kafka Info">
+        <Typography variant="body1">Error fetching topic data: No Topic Data</Typography>
+      </InfoCard>
+    )
+  }
+
+  const columns: TableColumn[] = [
+    { title: 'Topic', field: 'topic_id'},
+    { title: 'Current Lag', field: 'topic_lag'},
+  ];
+
+  const data = topics.items.map(inc => {
+    return {
+      topic_id: inc.metric.topic,
+      topic_lag: inc.value.at(-1)
+    };
+  });
+
+  return (
+    <InfoCard>
+      <Table
+        title="Topic Lag"
+        options={{
+          search: false,
+          paging: true,
+          pageSize: 5,
+          padding: 'dense',
+        }}
+        columns={columns}
+        data={data}
+        isLoading={loading}
+      />
+    </InfoCard>
+  );
+};
 
 export function KafkaInfoComponent() {
   const { entity } = useEntity();
-  const title = 'Kafka Information';
 
   // Get the config object from backstage
   const config = useApi(configApiRef);
 
-  const [loading, setLoading] = useState(true);
-  const [error, setError] = useState(false);
-
-  // Set up some state info for the response from the backend
-  const [metricResponse, setMetricResponse] = useState<Object>({});
-  const [filteredResponse, setFilteredResponse] = useState<Object>({});
+  // Set up backstage fetchApi for authenticated routes
+  const fetchApi = useApi(fetchApiRef);
 
-  // Get backend URL from the config
-  const backendUrl = config.getString('backend.baseUrl');
+  // Get cluster config
+  const [clusterMap, _setClusterMap] = useState(config.get('kafkaInfo.clusters'));
+  const [currentClusterIdx, setCurrentClusterIdx] = useState(0);
 
   // Get defined consumer group from entity
   const consumerGroup = entity.metadata.annotations?.[KAFKA_INFO_ANNOTATION].split(',') ?? '';
 
-  useEffect(() => {
-    setLoading(true);
-    // Directly query a prometheus endpoint for metric data
-    fetch(`${backendUrl}/api/proxy/kafka-lag/query?query=aws_kafka_max_offset_lag_sum`)
-      .then(response => {
-        return response.json();
-      })
-      .then(text => {
-        setMetricResponse(text);
-      })
-      .catch(error => {
-        setError(true);
-        console.error('Error fetching topic data:', error);
-        setLoading(false);
-      });
-  }, []);
-
-  useEffect(() => {
-    setLoading(true);
-    setError(false);
-    const filteredGroup = metricResponse.data?.result?.filter((mentry) => {
-      return consumerGroup.some(e => { return e === mentry.metric.group });
-    });
-    if (filteredGroup == "" || filteredGroup === undefined) {
-      setError(true);
-    }
-    setFilteredResponse(filteredGroup);
-    setLoading(false);
-  }, [metricResponse]);
+  const { value, loading, error } = useAsync(async (): Promise<
```

Comment:
this isn't wrong, but its odd. We do fetches in normal side effects or functions called by side effects all the time. I'm curious why this is implemented like this. It's OK, maybe even good because you don't have to manage loading state manually, but why did you reach for this.

Also, this fetch needs more robust error handling. Right now if you get something you don't expect back from the server, like HTML rather than json the attempt to parse the JSON bombs out and crashes the web app in the client browser.